src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -233,25 +233,25 @@ public QuorumCnxManager(QuorumPeer self) {
         listener = new Listener();
         listener.setName("QuorumPeerListener");
     }
-
+    
     /**
-     * Invokes initiateConnection for testing purposes
-     * 
+     * establish Connection with sid using its electionAddr.
      * @param sid
+     * @param sock
+     * @throws IOException
      */
-    public void testInitiateConnection(long sid) throws Exception {
-        LOG.debug("Opening channel to server " + sid);
-        Socket sock = new Socket();
+	public void establishConnection(Long sid, Socket sock) throws IOException {
+		LOG.debug("Opening channel to server " + sid);
         setSockOpts(sock);
         sock.connect(self.getVotingView().get(sid).electionAddr, cnxTO);
-        initiateConnection(sock, sid);
-    }
+        LOG.debug("Connected to server " + sid);
+	}
     
     /**
      * If this server has initiated the connection, then it gives up on the
      * connection if it loses challenge. Otherwise, it keeps the connection.
      */
-    public boolean initiateConnection(Socket sock, Long sid) {
+    public boolean initiateConnection(Long sid, Socket sock) {
```

Comment:
I think put the mutable variables after fixed variables may be better in the parameter list;
such as :
`private void myMethod(int a, Integer b, String s1, StringBuilder s2){}`